### PR TITLE
feat(react-dialog): Makes `DialogBody` responsive to large content

### PR DESCRIPTION
**Pull request by bsunderhus**
## Current Behaviour

Currently `DialogBody` has its height set to `fit-content` which allows it to stretch no matter how big internal content is, which actually breaks on cases the content is bigger them viewports size itself.

image

## New Behaviour

Updates `DialogBody` style to set its height to `100%` and making overflowed content scrollable.

image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                          |     Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                  Change |
| :------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | --------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr> | `84.804 kB`<br />`25.244 kB` | `84.793 kB`<br />`25.27 kB` | `-11 B` <br />`26 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.01 kB`<br />`51.922 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `32.876 kB`<br />`10.773 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/0547583ebcebe746110e5ef8d0599b0972b1f4c6'>0547583ebcebe746110e5ef8d0599b0972b1f4c6</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/Avatar-mount_base.html">1266</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/Avatar-mount.html">1259</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/Button-mount_base.html">963</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/Button-mount.html">980</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/FluentProvider-mount_base.html">1602</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/FluentProvider-mount.html">1582</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/FluentProviderWithTheme-mount_base.html">631</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/FluentProviderWithTheme-mount.html">641</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">593</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">605</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">643</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">649</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/MakeStyles-mount_base.html">1936</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/MakeStyles-mount.html">1941</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/SpinButton-mount_base.html">2510</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24354/perf-test-react-components/SpinButton-mount.html">2527</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-dialog/src/stories/Dialog/DialogResponsiveness.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody, DialogActions } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+
+export const ScrollingLongContent = () => {
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/stories/Dialog/DialogResponsiveness.stories.tsx:5]:
```suggestion
export const Responsiveness = () => {
```

nit: to match filename

Comment (layershifter) [reply packages/react-components/react-dialog/src/stories/Dialog/DialogResponsiveness.stories.tsx:5]: Or rename the file 🐱

**Review by layershifter — APPROVED** [review]

**Pull request merged**